src/query/frame_change.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <unordered_map>
+#include "query/typed_value.hpp"
+#include "utils/memory.hpp"
+#include "utils/pmr/unordered_map.hpp"
+
+namespace memgraph::query {
+
+using CachedType = std::unordered_map<size_t, std::vector<TypedValue>>;
```

Comment:
From `cppreference`, it seems to me that `multimap` is not a better solution than `std::unordered_map` as it does search in logarithmic time for a certain key. https://en.cppreference.com/w/cpp/container/multimap/equal_range 

The thing is we need to do an `equal_range` kind of search as it is not guaranteed by `multimap` which key would it return, so it is certainly slower than `unordered_map` which does a constant search for the key https://en.cppreference.com/w/cpp/container/unordered_map/find.

If you still think it makes sense to do these benchmarks, I can do them.

And `unordered_set` for `TypedValues` will use some kind of hash function and we are back to square one, aren't we? Because we exactly tried to do explicit check with == whether values are equal.

And I think unorderd_set will be slower as it needs to do always one more hash to hash them inside these buckets, and then check for equality. But with std::vector we do just one check for equality.

Reply:
Yeah, I think you are right.